sdk/src/wallet/account/operations/transaction/prepare_output.rs

```diff
@@ -97,263 +96,182 @@ where
             }
         }
 
-        let first_output = first_output_builder.finish(token_supply)?;
+        // Build output with minimum required storage deposit so we can use the amount in the next step
+        let first_output = first_output_builder
+            .with_minimum_storage_deposit(rent_structure)
+            .finish_output(token_supply)?;
 
-        let mut second_output_builder = BasicOutputBuilder::from(&first_output);
+        let mut second_output_builder = if nft_id.is_some() {
+            OutputBuilder::Nft(NftOutputBuilder::from(first_output.as_nft()))
+        } else {
+            OutputBuilder::Basic(BasicOutputBuilder::from(first_output.as_basic()))
+        };
 
-        // Update the amount
-        match params.amount.cmp(&first_output.amount()) {
-            Ordering::Greater | Ordering::Equal => {
-                // if it's larger than the minimum storage deposit, just replace it
-                second_output_builder = second_output_builder.with_amount(params.amount);
-            }
-            Ordering::Less => {
-                let storage_deposit = params.storage_deposit.unwrap_or_default();
-                // Gift return strategy doesn't need a change, since the amount is already the minimum storage
-                // deposit
-                if storage_deposit.return_strategy.unwrap_or_default() == ReturnStrategy::Return {
-                    let remainder_address = self.get_remainder_address(transaction_options).await?;
+        let min_storage_deposit_basic_output =
+            MinimumStorageDepositBasicOutput::new(rent_structure, token_supply).finish()?;
 
-                    // Calculate the minimum storage deposit to be returned
-                    let min_storage_deposit_return_amount =
-                        MinimumStorageDepositBasicOutput::new(rent_structure, token_supply).finish()?;
+        if params.amount > min_storage_deposit_basic_output {
```

Comment:
Just checking, what if they are equal? because you have > and then later <

Reply:
Then the value just stays, basically the same as `>=`, just that it's not set again